metron-analytics/metron-statistics/src/main/java/org/apache/metron/statistics/MathFunctions.java

```diff
@@ -59,4 +60,45 @@ public boolean isInitialized() {
       return true;
     }
   }
+
+  /**
+   * Calculates the statistical bin that a value falls in.
+   */
+  @Stellar(name = "BIN"
+          , description = "Computes the bin that the value is in given a set of bounds."
+          , params = {
+           "value - The value to bin"
+          , "bounds - A list of value bounds (excluding min and max) in sorted order."
+                    }
+          ,returns = "Which bin N the value falls in such that bound(N-1) < value <= bound(N). " +
+          "No min and max bounds are provided, so values smaller than the 0'th bound go in the 0'th bin, " +
+          "and values greater than the last bound go in the M'th bin."
+  )
+  public static class Bin extends BaseStellarFunction {
+
+    public static int getBin(double value, int numBins, Function<Integer, Double> boundFunc) {
+      double lastBound = Long.MIN_VALUE;
```

Comment:
Double.NEGATIVE_INFINITY or -Double.MAX_VALUE would be better than Long.MIN_VALUE